### PR TITLE
Use go.mod instead of glide

### PR DESCRIPTION
**Pull request by SVilgelm**
Initialize go.mod
Remove glide files
Clean up Makefile
Remove unused `shuffle` function

### CODE REVIEW

**Comment by SVilgelm** [comment]
@wk8 please review

File: Makefile [hunk 1]
```diff
@@ -1,40 +1,6 @@
...
-$(GLIDE_BIN):
-ifeq ($(GOPATH),)
-	@ echo "GOPATH env var not defined, cannot install glide"
-	exit 1
-endif
-ifeq ($(WGET),)
-	$(CURL) $(GLIDE_URL) | sh
-else
-	$(WGET) -O - $(GLIDE_URL) 2> /dev/null | sh
-endif
```
Comment (wk8) [inline Makefile:40]: That makefile is quite simpler all of a sudden! :)

File: orderedmap_test.go [hunk 2]
```diff
@@ -261,16 +260,6 @@ func assertLenEqual(t *testing.T, om *OrderedMap, expectedLen int) {
 	assert.Equal(t, expectedLen, om.list.Len())
 }
 
-func shuffle(a []interface{}) {
```
Comment (wk8) [inline orderedmap_test.go:264]: Mmh good catch, the `TestShuffle` actually doesn't use this, dunno what I was thinking when I wrote it :/ I'll fix that later, thanks for noticing!

**Comment by SVilgelm** [comment]
thank you, I used `golangci-lint` tool with default configuration and it showed unused code

**Pull request merged**